packages/dataviews/src/view-list.tsx

```diff
@@ -150,12 +178,12 @@ export default function ViewList( {
 	);
 
 	const onSelect = useCallback(
-		( item ) => onSelectionChange( [ item ] ),
+		( item: Item ) => onSelectionChange( [ item ] ),
 		[ onSelectionChange ]
 	);
 
 	const getItemDomId = useCallback(
-		( item ) => ( item ? `${ baseId }-${ getItemId( item ) }` : undefined ),
+		( item?: Item ) => ( item ? `${ baseId }-${ getItemId( item ) }` : '' ),
```

Comment:
Returning an empty string instead of `undefined` changes the behavior of `getItemDomId` breaking the keyboard navigation. Fix at https://github.com/WordPress/gutenberg/pull/61478